src/main/java/org/truffleruby/core/hash/SetNode.java

```diff
@@ -22,6 +22,9 @@
 
 @ImportStatic(HashGuards.class)
 public abstract class SetNode extends RubyContextNode {
+    /* On execution SetNode performs Hash#[]= and returns true if the key is newly added to hash, and returns false if
+     * the key was found in the hash prior. This is different from the return of Hash#[]=, which returns the right hand
+     * side value, however this change allows get and set to happen concurrently. */
```

Comment:
`concurrently` seems a but confusing here, maybe "without an extra lookup"?

Reply:
"... however this change does a set and returns the result of get without an extra lookup." ?